devlib/module/gem5stats.py

```diff
@@ -66,6 +67,7 @@ def __init__(self, target):
         self._stats_file_path = os.path.join(target.platform.gem5_out_dir,
                                             'stats.txt')
         self.rois = {}
+        self._dump_pos_cache = {0: 0}
```

Comment:
Actually having a dict makes line 173 very simple to implement. Since a long run will involve a lot (thousands ?) of dumps not all of them are cached in `_dump_pos_cache`. For example, I will end-up with something like 
```
self._dump_pos_cache = { 
    0: 0,
    4: 123456,
    19: 7891234
}
```

if use `_goto_dump` on dumps 4 and 19. Putting this in a list will make things more difficult since the Xth dump in the list is not necessarily be the Xth dump in the file. If we really want to use a list, we probably want to cache each and every dump by modifying line 191 and putting it into the `while` loop. I you think it's better I'm OK with it, the only concern is maybe the memory overhead.